jetty-ee9/jetty-ee9-tests/jetty-ee9-test-jersey/src/test/java/org/eclipse/jetty/ee9/jersey/tests/HungBlockingThreadsTest.java

```diff
@@ -0,0 +1,230 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.not;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class HungBlockingThreadsTest
+{
+    private static final Logger LOG = LoggerFactory.getLogger(HungBlockingThreadsTest.class);
+    private static final int THREAD_COUNT = 128;
+    private Server server;
+    private ExecutorService executorService;
+    private HttpClient httpClient;
+
+    @BeforeAll
+    public static void beforeAll()
+    {
+        // Wire up java.util.logging to slf4j.
+        org.slf4j.bridge.SLF4JBridgeHandler.removeHandlersForRootLogger();
+        org.slf4j.bridge.SLF4JBridgeHandler.install();
+    }
+
+    @AfterAll
+    public static void afterAll()
+    {
+        org.slf4j.bridge.SLF4JBridgeHandler.uninstall();
+    }
+
+    @BeforeEach
+    public void setUp() throws Exception
+    {
+        QueuedThreadPool queuedThreadPool = new QueuedThreadPool(10, 10, (int)TimeUnit.MINUTES.toMillis(1000L));
+        queuedThreadPool.setName("server");
+        queuedThreadPool.setDaemon(true);
+        queuedThreadPool.setReservedThreads(1);
+
+        server = new Server(queuedThreadPool);
+
+        ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);
+        context.setContextPath("/");
+
+        ResourceConfig resourceConfig = new ResourceConfig();
+        Resource resource = Resource.builder(PostResource.class).build();
+        Resource.Builder resourceBuilder = Resource.builder(resource);
+        resource.getResourceMethods().forEach(resourceMethod -> resourceBuilder.updateMethod(resourceMethod).managedAsync());
+        resourceConfig.registerResources(resourceBuilder.build());
+        resourceConfig.register(CustomFeature.class);
+
+        context.addServlet(new ServletHolder(new ServletContainer(resourceConfig)), "/*");
+
+        server.setHandler(context);
+
+        ServerConnector connector = new ServerConnector(server);
+        connector.setPort(0);
+        connector.setIdleTimeout(Long.MAX_VALUE);
+        server.addConnector(connector);
+        server.start();
+
+        httpClient = new HttpClient();
+        QueuedThreadPool clientQtp = new QueuedThreadPool(THREAD_COUNT * 2);
+        clientQtp.setName("client");
+        httpClient.setExecutor(clientQtp);
+        httpClient.setMaxConnectionsPerDestination(THREAD_COUNT);
+        httpClient.start();
+
+        executorService = Executors.newCachedThreadPool();
+    }
+
+    @AfterEach
+    public void tearDown()
+    {
+        executorService.shutdownNow();
+        LifeCycle.stop(server);
+        LifeCycle.stop(httpClient);
+    }
+
+    @Test
+    public void test() throws Exception
```

Comment:
Annotate as "stress"?

Also maybe add a comment saying: "this test tries to reproduce the issue #... by running THREAD_COUNT request in the hope to reproduce the issue statistically."